modules/instrument_builder/jsx/react.questions.js

```diff
@@ -616,10 +616,18 @@ var AddElement = React.createClass({
         hasError = true;
       }
 
-      if (minDate > maxDate && min !== '' && max !== '') {
+      if (min === '' || max === '') {
         let temp = (this.state.error) ? this.state.error : {};
 
-        temp.dateOption = "End year append befor start year";
+        temp.dateOption = "Invalid start or end year";
+        this.setState({
+          error: temp
+        });
+        hasError = true;
+      } else if (minDate > maxDate) {
+        let temp = (this.state.error) ? this.state.error : {};
+
+        temp.dateOption = "End year append before start year";
```

Comment:
perhaps reword this to say "End date is earlier than start date" or "End date is before start date"

Reply:
I think it's better to use "End date is before start date" than the previous choice. Fixed.